src/DeviceInterfaceValues.tsx

```diff
@@ -184,6 +189,346 @@ const InterfaceData = ({ interfaceData }: InterfaceDataProps): React.ReactElemen
   );
 };
 
+interface SendInterfaceDataModalProps {
+  showModal: boolean;
+  interfaceData: AstarteInterface | undefined;
```

Comment:
This is not data that has to be sent to the interface, but rather, the whole interface definition. I suggest renaming that to `interfaceDefinition` to avoid mistakes.

Also if the interface definition was not loaded, we can't display anything to the user, so this cannot be `undefined`